Dockerfile

```diff
@@ -1,9 +1,16 @@
 FROM hseeberger/scala-sbt:8u242_1.3.7_2.12.10
 
 WORKDIR /opt/book-api
+EXPOSE 9000
 
-ADD . /opt/book-api
+# add build and project so that dependencies can be downloaded -- this will only rerun if the dependencies change,
+# but not if the code changes
+ADD ./project   ./project
+ADD ./build.sbt ./build.sbt
+RUN sbt update
```

Comment:
this is downloads all the dependencies, similar to adding `package.json` and then running `npm install`